CMakeLists.txt

```diff
@@ -0,0 +1,427 @@
+set(FMTDYTST code/fmtdy.c code/fmtno.c code/fmtdytst.c)
+set(FMTHETST code/fmthe.c code/fmtdy.c code/fmtno.c code/fmtdytst.c)
+set(FMTSCM code/fmtscheme.c)
+set(PLINTH code/mpsliban.c code/mpsioan.c)
+set(MPMCOMMON
+    code/abq.c
+    code/arena.c
+    code/arenacl.c
+    code/arenavm.c
+    code/arg.c
+    code/boot.c
+    code/bt.c
+    code/buffer.c
+    code/cbs.c
+    code/dbgpool.c
+    code/dbgpooli.c
+    code/event.c
+    code/failover.c
+    code/format.c
+    code/freelist.c
+    code/global.c
+    code/land.c
+    code/ld.c
+    code/locus.c
+    code/message.c
+    code/meter.c
+    code/mpm.c
+    code/mpsi.c
+    code/nailboard.c
+    code/policy.c
+    code/pool.c
+    code/poolabs.c
+    code/poolmfs.c
+    code/poolmrg.c
+    code/protocol.c
+    code/range.c
+    code/rangetree.c
+    code/ref.c
+    code/ring.c
+    code/root.c
+    code/sa.c
+    code/sac.c
+    code/scan.c
+    code/seg.c
+    code/shield.c
+    code/splay.c
+    code/ss.c
+    code/table.c
+    code/trace.c
+    code/traceanc.c
+    code/tract.c
+    code/tree.c
+    code/version.c
+    code/vm.c
+    code/walk.c
+)
+set(POOLS ${AMC} ${AMS} ${AWL} ${LO} ${MV2} ${MVFF} ${SNC})
+set(MPM ${MPMCOMMON} ${MPMPF} ${POOLS} ${PLINTH})
+
+
+# Varieties
+#
+# MPS varieties (HOT, COOL, etc.) probably map to CMake "Build
+# Configurations"
+# <https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#build-configurations>.
+#
+# FIXME: I'm just mapping variety.cool (Debug) and variety.hot
+# (Release) for now, but we need to add variety.rash at least for
+# automatic comparison with hot.  It would be good to see if
+# variety.cold builds and runs too.
+
+add_compile_definitions(
+  $<$<CONFIG:Debug>:CONFIG_VAR_COOL>
+  $<$<CONFIG:Release>:CONFIG_VAR_HOT>
+)
+
+# Override CMake's default flags because we specify all flags.
+message(VERBOSE "Setting CMAKE_C_FLAGS from ${CMAKE_C_FLAGS} to ${CFLAGSCOMPILER}")
+list(JOIN CFLAGSCOMPILER " " CMAKE_C_FLAGS)
+message(VERBOSE "Setting CMAKE_C_FLAGS_DEBUG from ${CMAKE_C_FLAGS_DEBUG} to ${CFLAGSDEBUG}")
+list(JOIN CFLAGSDEBUG " " CMAKE_C_FLAGS_DEBUG)
+message(VERBOSE "Setting CMAKE_C_FLAGS_RELEASE from ${CMAKE_C_FLAGS_RELEASE} to ${CFLAGSOPT}")
+list(JOIN CFLAGSOPT " " CMAKE_C_FLAGS_RELEASE)
+
+
+# Targets
+#
+# Just using gcbench as an example target for now.
+
+#add_library(mps OBJECT code/mps.c)
+#target_compile_options(mps PRIVATE ${CFLAGSCOMPILERSTRICT})
+
+add_library(mps STATIC
+  $<$<CONFIG:Release>:code/mps.c>
+  $<$<CONFIG:Debug>:${MPM}>
+)
+target_compile_options(mps PRIVATE ${CFLAGSCOMPILERSTRICT})
+
+add_library(mpsplan STATIC ${PLINTH})
+target_compile_options(mpsplan PRIVATE ${CFLAGSCOMPILERSTRICT})
+
+add_executable(gcbench code/gcbench.c ${FMTDYTST} ${TESTLIB} ${TESTTHR})
+target_compile_options(gcbench PRIVATE ${CFLAGSCOMPILERSTRICT})
+# FIXME: Should be C89 but CMake doesn't know it.
+# FIXME: How to set this for all targets not just gcbench?
+# target_compile_features(gcbench PRIVATE c_std_90)
```

Comment:
Towards the top of the file, you can do:

```
SET(CMAKE_C_STANDARD 90)
```

That will set it for all subsequent targets.